python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/scenarios/auto_materialize_policy_scenarios.py

```diff
@@ -531,4 +559,84 @@ def with_auto_materialize_policy(
+        assets=with_auto_materialize_policy(
+            vee,
+            AutoMaterializePolicy.eager().with_rules(
+                AutoMaterializeRule.skip_on_not_all_parents_updated(),
+            ),
+        ),
+        cursor_from=AssetReconciliationScenario(
+            assets=with_auto_materialize_policy(
+                vee,
+                AutoMaterializePolicy.eager().with_rules(
+                    AutoMaterializeRule.skip_on_not_all_parents_updated(),
+                ),
+            ),
+            unevaluated_runs=[run(["A", "B", "C"])],
+            expected_run_requests=[],
+        ),
+        unevaluated_runs=[run(["A"]), run(["B"]), run(["A"])],
+        expected_run_requests=[run_request(["C"])],
+    ),
+    "test_wait_for_all_parents_updated_unpartitioned_with_one_parent_partitioned": AssetReconciliationScenario(
+        assets=unpartitioned_with_one_parent_partitioned_skip_on_not_all_parents_updated,
+        cursor_from=AssetReconciliationScenario(
+            assets=unpartitioned_with_one_parent_partitioned_skip_on_not_all_parents_updated,
+            cursor_from=AssetReconciliationScenario(
+                assets=unpartitioned_with_one_parent_partitioned_skip_on_not_all_parents_updated,
+                unevaluated_runs=[run(["asset2"]), run(["asset1"], partition_key="2020-01-01")],
+                current_time=create_pendulum_time(year=2020, month=1, day=3, hour=1),
+                expected_run_requests=[],
+            ),
+            unevaluated_runs=[run(["asset1"], "2020-01-02")],
+            current_time=create_pendulum_time(year=2020, month=1, day=3, hour=1),
+            expected_run_requests=[run_request(["asset3"])],
+        ),
+        unevaluated_runs=[
+            run(["asset1"], "2020-01-01"),
+            run(["asset1"], "2020-01-02"),
+            run(["asset2"]),
+        ],
+        current_time=create_pendulum_time(year=2020, month=1, day=3, hour=1),
+        expected_run_requests=[run_request(["asset3"])],
+    ),
+    "test_wait_for_all_parents_updated_one_upstream_starts_later_than_downstream": AssetReconciliationScenario(
+        assets=one_upstream_starts_later_than_downstream_skip_on_not_all_parents_updated,
+        cursor_from=AssetReconciliationScenario(
+            assets=one_upstream_starts_later_than_downstream_skip_on_not_all_parents_updated,
+            cursor_from=AssetReconciliationScenario(
+                assets=one_upstream_starts_later_than_downstream_skip_on_not_all_parents_updated,
+                unevaluated_runs=[run(["asset2"], partition_key="2023-01-01-00:00")],
+                current_time=create_pendulum_time(year=2023, month=1, day=1, hour=4),
+                expected_run_requests=[run_request(["asset3"], partition_key="2023-01-01-00:00")],
+            ),
+            unevaluated_runs=[run(["asset1"], "2023-01-01-03:00")],
+            current_time=create_pendulum_time(year=2023, month=1, day=1, hour=4),
+            expected_run_requests=[],
```

Comment:
Do we expect asset2 to have a run request for the `2023-01-01-03:00` partition here? I thought it would, because it doesn't have upstreams and this partition is the latest partition that is unmaterialized

Reply:
hm yeah I think I agree that I'd expect a run request there